src/components/Common/CopyButton.tsx

```diff
@@ -0,0 +1,54 @@
+import { VariantProps } from "class-variance-authority";
+import { useState } from "react";
+import * as React from "react";
+
+import CareIcon from "@/CAREUI/icons/CareIcon";
+
+import { Button, buttonVariants } from "@/components/ui/button";
+import { TooltipComponent, TooltipProvider } from "@/components/ui/tooltip";
+
+import { copyToClipboard } from "@/Utils/utils";
+
+export interface CopyButtonProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
+    VariantProps<typeof buttonVariants> {
+  children?: React.ReactNode;
+  content: string | undefined;
+  tooltipContent?: string;
+}
+
+const CopyButton = ({
+  content,
+  tooltipContent = "Copy to clipboard",
+  children,
+  size,
+}: CopyButtonProps) => {
+  const [isCopied, setIsCopied] = useState(false);
+
+  if (content === undefined) return null;
+
+  return (
+    <TooltipProvider>
+      <TooltipComponent content={isCopied ? "Copied!" : tooltipContent}>
```

Comment:
do follow the merge checklist in the PR comment.

add to i18n strings.

refer: https://github.com/ohcnetwork/care_fe?tab=readme-ov-file#translations